listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -621,27 +621,31 @@ def import_listens_dump(self, archive_path: str, threads: int = DUMP_DEFAULT_THR
 
         return total_imported
 
-    def delete(self, user_id):
+    def delete(self, user_id, created=None):
         """ Delete all listens for user with specified user ID.
 
         Note: this method tries to delete the user 5 times before giving up.
 
         Args:
             musicbrainz_id: the MusicBrainz ID of the user
             user_id: the listenbrainz row id of the user
+            created: delete listens created before this timestamp
 
         Raises: Exception if unable to delete the user in 5 retries
         """
-        query = """
+        if created:
```

Comment:
is this right? If it is not null (should be 'is not None'?) then the value is overwritten?

Reply:
:facepalm: , that is wrong. Thanks for the catch. I'll fix.

Reply:
done